networkit/cpp/embedding/AliasSampler.cpp

```diff
@@ -0,0 +1,87 @@
+/*
+ * AliasSampler.cpp
+ *
+ *  Created on: 03.07.2020
+ *      Author: Klaus Ahrens  <ahrens@informatik.hu-berlin.de>
+ *
+ *  adapted and reimplemented in C++17
+ *  from node2vec [https://arxiv.org/pdf/1607.00653v1.pdf]
```

Comment:
What do you mean by "adapted"? Is this based on existing code?

Reply:
Klaus did re-implement the algorithm (without copying the code literally, as far as I understand). However, since the structure still matches the original code, it is probably a good idea to acknowledge the copyright. Luckily, this is not an issue since the original code is also under MIT.